core/src/main/java/bio/terra/pearl/core/service/export/EnrolleeExportService.java

```diff
@@ -222,32 +214,47 @@ protected List<SurveyFormatter> generateSurveyModules(ExportOptions exportOption
         return moduleFormatters;
     }
 
-    protected List<EnrolleeExportData> loadEnrolleesForExport(Study study,
-                                                              StudyEnvironmentConfig config,
-                                                              List<Enrollee> enrollees) {
-        // for now, load each enrollee individually.  Later we'll want more sophisticated batching strategies
-        return enrollees
-                .stream()
-                .map(enrollee -> loadEnrolleeData(study, config, enrollee))
+    public List<EnrolleeExportData> loadEnrolleeExportData(UUID studyEnvironmentId, ExportOptionsWithExpression exportOptions) {
+        Study study = studyService.findByStudyEnvironmentId(studyEnvironmentId).orElseThrow();
+        List<Enrollee> enrollees = loadEnrollees(studyEnvironmentId, exportOptions.getFilterExpression(), exportOptions.getRowLimit());
+
+        List<UUID> enrolleeIds = enrollees.stream().map(Enrollee::getId).toList();
+        List<UUID> profileIds = enrollees.stream().map(Enrollee::getProfileId).toList();
+        List<UUID> participantUserIds = enrollees.stream().map(Enrollee::getParticipantUserId).toList();
+
+        // batch load the following modules to reduce the number of queries and reduce the memory footprint of data exports.
+        // eventually, the in-clauses of these sql queries will be too large, and we'll need to batch load these in smaller chunks
```

Comment:
just to make sure I understand, when you say batch load, is the idea to refactor our export like so?
- create CSV/TSV object early/at beginning
- load, e.g. 100 enrollees, add to CSV
- load next 100 enrollees, add to CSV, etc. until we're finished?

Reply:
the key idea would be to just limit the in-clauses to some number of enrollee ids (we can probably get away with an in-clause containing a few thousand ids)

how we decide to assemble it all is still TBD. once we get to the point of having to worry about in-clause size limits, we probably want to start thinking about streaming the tsv back to the user as we assemble it (and that might look a lot like what you described)